apps/client/app/purchase-orders/[id]/page.tsx

```diff
@@ -1,36 +1,86 @@
+'use client';
 import Error from '@/components/common/error';
 import {
   safePurchaseOrderId,
   PurchaseOrderService,
+  PurchaseOrderFull,
 } from '@/lib/services/purchase-orders.service';
 import {Result} from '@/lib/types/result';
 import {notFound} from 'next/navigation';
 import Detail from './components/detail';
+import {useEffect, useState} from 'react';
+import Loading from '../loading';
+import {PermissionGuard} from '@/components/permission-guard';
+import {useAuth} from '@/lib/contexts/auth-context';
 
 interface Props {
   params: {id: string};
 }
 
-export default async function Page(props: Props) {
+export default function Page(props: Props) {
+  const {user} = useAuth();
+  const [isMounted, setIsMounted] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [purchaseOrderRes, setPurchaseOrderRes] = useState<Result<
+    PurchaseOrderFull,
+    string
+  > | null>(null);
+
   const idRes = safePurchaseOrderId(props.params.id);
   if (Result.isErr(idRes)) {
     notFound();
   }
 
   const id = idRes.data;
-  const purchaseOrderRes = await PurchaseOrderService.findOne(id);
+
+  useEffect(() => {
+    setIsMounted(true);
+  }, []);
+
+  useEffect(() => {
+    if (!isMounted) return;
+
+    const fetchPurchaseOrder = async () => {
+      setLoading(true);
+      const result = await PurchaseOrderService.findOne(id);
+      setPurchaseOrderRes(result);
+      setLoading(false);
+    };
+
+    fetchPurchaseOrder();
+  }, [id, isMounted, user]);
```

Comment:
Tell me about the `useEffect` dependency array? Why is `user` included here?

Reply:
This was mostly for demo purposes (you'll see in the video) but when a user first creates an account, they don't have any roles. I have a demo endpoint to add a role to a user, then a demo UI feature to refresh the token from the server, and wanted this component to re-render when the new user token was fetched.